packages/react-component-library/src/forms/vanilla/useVanillaForm.ts

```diff
@@ -0,0 +1,99 @@
+import React, { useState } from 'react'
+
+import { sleep } from '../../helpers'
+
+interface FormErrors {
+  [key: string]: boolean
+}
+
+interface SyntheticFormEvent {
+  target: {
+    name: string
+    value: string | string[] | number | number[]
+  }
+}
+
+export const useVanillaForm = <T>(
+  initialState: T,
+  initialErrors: FormErrors,
+  validation: Record<string, (value: string) => boolean>
+) => {
+  const [formErrors, setFormErrors] = useState<FormErrors>(initialErrors)
+  const [formPayload, setFormPayload] = useState<T | undefined>()
+  const [formState, setFormState] = useState<T>(initialState)
+  const [isSubmitting, setIsSubmitting] = useState<boolean>(false)
+
+  const handleChange = (e: React.FormEvent | SyntheticFormEvent): void => {
+    const { name, value } = e.target as HTMLInputElement
```

Comment:
Are we able to use `React.FormEvent<HTMLInputElement>` and `e.currentTarget` here if `SyntheticFormEvent` is changed to match?